python/gui/MaaS/cbv/MapView.py

```diff
@@ -34,52 +36,100 @@ def _build_fabric_path(fabric, type):
         resolution = resolution_match.group(2)
     else:
         name = fabric
-        resolution=''
+        resolution = ''
+
+    hyfab_data_dir = Path(PROJECT_ROOT, 'static', 'ngen', 'hydrofabric', name, resolution)
+
+    geojson_file = hyfab_data_dir.joinpath(f"{type}_data.geojson")
+    if geojson_file.exists():
+        return geojson_file
+
+    if hyfab_data_dir.joinpath("hydrofabric.gpkg").exists():
+        geopackage_file = hyfab_data_dir.joinpath("hydrofabric.gpkg")
+    elif hyfab_data_dir.joinpath(f"{name}.gpkg").exists():
+        geopackage_file = hyfab_data_dir.joinpath(f"{name}.gpkg")
+    else:
+        logger.error(f"Can't build fabric path: can't find hydrofabric data file in directory {hyfab_data_dir!s}")
+        return None
+
+    return geopackage_file
 
-    path = Path(PROJECT_ROOT, 'static', 'ngen', 'hydrofabric', name, resolution, type+'_data.geojson')
-    return path
 
 class Fabrics(APIView):
     def get(self, request: HttpRequest, fabric: str = None) -> typing.Optional[JsonResponse]:
         if fabric is None:
-            fabric = 'example'
+            fabric = 'example_fabric_name'
         type = request.GET.get('fabric_type', 'catchment')
         if not type:
-            type="catchment"
+            type = "catchment"
```

Comment:
It might be wise to abstract out this logic into another function that would be somewhat "easy" to swap in and out. Won't the fabrics come via the data store in the future?